File: blog/src/main/java/cn/pompip/myblog/control/ArticleController.java
package cn.pompip.myblog.control;

import cn.pompip.myblog.entity.ArticleEntity;
import cn.pompip.myblog.model.WebPage;
import cn.pompip.myblog.server.ArticleServer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/article")
//@CrossOrigin
public class ArticleController {
    private ArticleServer server;

    @Autowired
    public ArticleController(ArticleServer server) {
        this.server = server;
    }

    @GetMapping("/all")
    public List<ArticleEntity> getArticleList() {
        return server.getAllArticle();
    }

    @GetMapping("/page/{page}")
    public WebPage<ArticleEntity> getArticleWithPage(@PathVariable int page) {
        return server.getArticleListWithPage(page);
    }

    @GetMapping("/detail/{id}")
    public ArticleEntity getArticle(@PathVariable long id) {
        return server.getArticle(id);
    }

    @PostMapping("/save")
    public ArticleEntity saveArticle( @RequestBody ArticleEntity articleEntity) {
        return server.saveArticle(articleEntity);
    }


    @GetMapping("/delete/{id}")
    public void deleteArticle(@PathVariable long id) {
        server.deleteArticle(id);
    }
}

File: src/main/kotlin/cn/pompip/myblog/control/ArticleController.kt
package cn.pompip.myblog.control

import cn.pompip.myblog.entity.ArticleEntity
import cn.pompip.myblog.server.ArticleServer
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/article")
class ArticleController {

    @Autowired
    lateinit var server: ArticleServer

    @PostMapping("/list")
    fun getArticleList(): List<ArticleEntity> = server.getIndexArticleList()


    @PostMapping("/{id}")
    fun getArticle(@PathVariable id: Long) = server.getOne(id)

    @PostMapping("/save")
    fun saveArticle(content: String) = server.saveArticle(content).id


    @PostMapping("/update")
    fun updateArticle(content: String, id: Long) = server.updateArticle(content, id).id

    @PostMapping("/delete/{id}")
    fun deleteArticle( @PathVariable id: Long) = server.deleteArticle(id)
}